docs/content/theming.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Theming
+---
+
+Theming in Primer React is made possible by a theme object that allows you to define your application's colors, spacing, fonts, and more.
+
+The structure of the [default theme object](/theme-reference) adheres to the [System UI Theme Specification](https://system-ui.com/theme) and contains the following keys:
+
+- colors
+- space
+- fonts
+- fontSizes
+- fontWeights
+- lineHeights
+- borderWidths
+- radii
+- shadows
+- breakpoints
+- sizes
+
+## ThemeProvider
+
+To give components access to the theme object, you must add `ThemeProvider` to the root of your application:
+
+```jsx
+import {ThemeProvider} from '@primer/components'
+
+function App() {
+  return (
+    <ThemeProvider>
+      <div>...</div>
+    </ThemeProvider>
+  )
+}
+```
+
+## Customizing the theme
+
+`ThemeProvider` accepts a `theme` prop that allows you to provide your own custom theme:
+
+```jsx
+import {ThemeProvider, theme} from '@primer/components'
+import deepmerge from 'deepmerge'
+
+const customTheme = deepmerge(theme, {
+  fonts: {
+    mono: 'Monolisa, monospace'
+  }
+})
+
+function App() {
+  return (
+    <ThemeProvider theme={customTheme}>
+      <div>...</div>
+    </ThemeProvider>
+  )
+}
+```
+
+Some components may break if your custom theme does not include all the same keys as the [default theme](/theme-reference). For that reason, we recommend extending the default theme using [deepmerge](https://www.npmjs.com/package/deepmerge).
+
+## Referencing theme values
+
+You can reference theme values in your application using [system props](/system-props), the [`sx` prop](/overriding-styles), the `themeGet` function, or the `useTheme` hook.
+
+### System props and the `sx` prop
+
+Some [system props](/system-props) and [`sx` prop](/overriding-styles) keys map to specific theme keys, allowing you to reference theme values in the corresponding object. For example, the `bg` prop maps to the `colors` theme key which means you can use the `bg` prop to reference values in the `colors` object:
+
+```jsx
+const theme = {
+  colors: {
+    bg: {
+      primary: '#fff'
+    }
+  }
+}
+
+function App() {
+  return (
+    <ThemeProvider theme={theme}>
+      <Box bg="bg.primary"></Box>
+      <Box sx={{bg: 'bg.primary'}}></Box>
+    </ThemeProvider>
+  )
+}
+```
+
+See the [Styled System Reference Table](https://styled-system.com/table) for a complete list of mappings.
+
+### themeGet
+
+```js
+import {themeGet} from '@primer/components'
+import styled from 'styled-components'
+
+const Example = styled.div`
+  background-color: ${themeGet('colors.bg.primary')};
+`
+```
+
+### useTheme
+
+```js
+import {useTheme} from '@primer/components'
+
+function Example() {
+  const theme = useTheme()
+  // theme.colors.bg.primary
+}
+```
```

Comment:
I'm assuming there is an advantage of using this over `themeGet` otherwise we probably wouldn't have two ways of doing the same thing, can you call that out here?